openreview/journal/process/remind_invited_reviewer_process.py

```diff
@@ -0,0 +1,50 @@
+def process(client, invitation):
+
+    print('Remind invited reviewers')
+    journal = openreview.journal.Journal()
+    grouped_edges = client.get_grouped_edges(invitation=journal.get_reviewer_invite_assignment_id(), label='Invitation Sent', groupby='id')
+    
+    if len(grouped_edges) == 0:
+        return
+    
+    edges = [openreview.api.Edge.from_json(edge['values'][0]) for edge in grouped_edges]
+
+    start_reminder_period = openreview.tools.datetime_millis(datetime.datetime.utcnow() - datetime.timedelta(weeks = journal.invite_assignment_reminder_period, days=1))
+    end_reminder_period = openreview.tools.datetime_millis(datetime.datetime.utcnow() - datetime.timedelta(weeks = journal.invite_assignment_reminder_period))
+    one_week = datetime.timedelta(weeks=1)
+    
+    print(f'reminder period between {start_reminder_period} and {end_reminder_period}')
+    for edge in edges:
+        if start_reminder_period < edge.tcdate and edge.tcdate < end_reminder_period:
+
+            notes=client.get_notes(id=edge.head)
+            if not notes:
+                continue
+            submission=notes[0]
+            user = edge.tail
+            user_profile=openreview.tools.get_profile(client, user)
+
+            if not user_profile:
+                user_profile=openreview.Profile(id=user,
+                    content={
+                        'names': [],
+                        'emails': [user],
+                        'preferredEmail': user
+                    })
+                
+            last_modified = datetime.timedelta(milliseconds=edge.tmdate - edge.tcdate)
+
+            if last_modified < one_week:
```

Comment:
I'm not sure why you need to make this check. Can you please explain?

Reply:
I agree that checking the 7 and 8 day interval is enough, but I am worried that if the date process fails in the middle and we re-run it, many reminders will be sent again. I am using the tmdate here as a tracker to know if the reminder has been sent or not.

Reply:
OK, I understand. Just one more question:
if this is true:
```py
if start_reminder_period < edge.tcdate and edge.tcdate < end_reminder_period:
```
isn't this false?
```py
if last_modified < one_week:
```

Reply:
`last_modified` is checking the difference between the tmdate and the tcdate, and I am updating the tmdate after I send the reminder. This should be happening about a week after the edge was created.